### DISCUSSION TITLE
(UI) Improved wiring controls

### DISCUSSION THREAD

**Discussion by Saevon** [FakeFishGames/Barotrauma#4711] [Ideas and suggestions]
# Problem

Wiring requires a lot of inputs to make simple logic connections:
 * Constantly selecting a new wire, having to click-drag the wire from its spawned location in Wiring-Panel
 * Rewiring from an old machine requiring drop in Wire-Panel, then find and double-click again in the sub-view to reselect it, before finally going to the new machine...
 * All or nothing approach, hard to pause a rewiring when you realize something is missing (input OR output)
 * Not always having context (you can only edit the current wire. the other wires all become non-interactable)

# Request List

- [ ]  1. Dragging out pulls the wire (auto selecting it for rewiring). This means you can go straight to a new machine and press E to move the wire quickly
- [x]  2. Deleting the wire is an explicit shortcut (or drag to side panel). (e.g like ask-4) Or a small trashcan at the bottom of the panel you can drop the wire to.
- [ ] 3. Opening a Wiring Panel while holding a wire should already select that wire.  So you just need to click on the selected in/out, rather then drag it from the bottom of the wiring panel
- [x] 4. When wiring (Sub-editor) there should be a shortcut to select the wire colour (1-8).
- [ ] 5. Right Click should throw away the current wire in Sub-Editor. (similar to item selection)
- [ ] 6. Hovering on wires while adding a new one. I need to see where existing wires go! (e.g was `relay_in_1` or `relay_in_2` the correct signal path?)
- [ ] 7. Dropping a detached wire without finishing it (on a square).  Letting you come back to that wire once the missing machinery exists. (Including the start of a wire!
- [ ] 8. Machinery placed near floating wires, shows the wires in the panel below. (Either directly OVER, or maybe 1 tile beside?)
- [ ]  9. Hover/select a component in wiring view, it should dim (alpha?) ALL unconnected wires, bringing the connected wires to the front
- [ ] 10.  Selecting a wire should dim all unrelated wires, and always move that one to the front for display 
- [ ] 11.  Selecting two components in build mode currently auto selects the wires,,, meaning you cannot recolour, or change transparency without fucking up your wire's colours too. It shouldn't do that…

### Circuit Box
- [ ] 1. When wiring (Circuit Box) there should be a shortcut to select the wire colour (1-8).

Edit: added more, renumbered for clarity, rephrased ask… late nights do not make good requests

**Comment by 3e849f2e5c** [comment]
> When you pull a wire out of the Rewiring View, it deletes it. Sometimes you want to move the wire to another machine instead.

Are you talking about in sub editor? Because in sub editor you can disconnect a wire from a pin and exit the UI and double click the loose end to pick it back up and wire it to something else

> When wiring (Sub-editor) there should be a shortcut to select the wire colour (1-8).

I implemented this by a request of someone else recently, you should see it the next update

**Reply by Saevon** [reply]
Ok, I've rephrased the discussion, hopefully its way more clear!

> disconnect a wire from a pin and exit the UI and double click the loose end to pick it back up and wire it to something else

This is exactly my problem, its doable… but non-intuitive (took me a while to discover) and quite hard to do. Removing wires is prioritized?

> I implemented this by a request of someone else recently, you should see it the next update

Love the second part! This will help so much

**Reply by 3e849f2e5c** [reply]
I'll keep the points in mind for the future, wiring in this game is a spaghetti monster I've yet to tame and I've only dealt with it in sub editor so far

**Reply by leonard84** [reply]
>>    disconnect a wire from a pin and exit the UI and double click the loose end to pick it back up and wire it to something else
>
>This is exactly my problem, its doable… but non-intuitive (took me a while to discover) and quite hard to do. Removing wires is prioritized?

How are you supposed to figure this out? I googled it didn't find anything, now by going over the discussions here one by one it is mentioned as a side note. I agree with @Saevon that this definite improvement. 

And it still leaves no way to do this in game.

>  1. Dragging out pulls the wire (auto selecting it for rewiring). This means you can go straight to a new machine and press E to move the wire quickly

Sound like a good and intuitive solution, it was actually what I tried to do quite often and was surprised that this would delete the wire in the editor and wouldn't do anything in game.
